File: 532. K-diff Pairs in an Array/Python/Solution.py
class Solution(object):
    def findPairs(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        if k < 0:
            return 0
        counts = {}
        for num in nums:
            if num not in counts:
                counts[num] = 0
            counts[num] += 1
        result = 0
        for key, value in counts.items():
            if k == 0:
                if value >= 2:
                    result += 1
            else:
                if key + k in counts:
                    result += 1
        return result

File: 532. K-diff Pairs in an Array/C++/Solution.cpp
class Solution {
public:
    int findPairs(vector<int> &nums, int k) {
        if (k < 0) {
            return 0;
        }
        unordered_map<int, int> counts;
        for (auto num : nums) {
            counts[num]++;
        }
        int result = 0;
        for (auto &item : counts) {
            if (k == 0) {
                if (item.second >= 2) {
                    result++;
                }
            } else {
                if (counts.find(item.first + k) != counts.end()) {
                    result++;
                }
            }
        }
        return result;
    }
};